front/app/components/SignUpIn/SignIn/PasswordSignin.tsx

```diff
@@ -159,53 +155,77 @@ class PasswordSignin extends PureComponent<
     }
   };
 
-  validate(email: string | null) {
+  validate(
+    phoneLoginEnabled: boolean,
+    emailOrPhoneNumber: string | null,
+    password: string | null
+  ) {
     const {
       intl: { formatMessage },
-      tenant,
     } = this.props;
-    const { password } = this.state;
-    const phone =
-      !isNilOrError(tenant) && tenant.attributes.settings.password_login?.phone;
-    const hasEmailError = !phone && (!email || !isValidEmail(email));
-    const emailError = hasEmailError
-      ? formatMessage(messages.emailError)
-      : null;
-    const hasEmptyPasswordError = !password;
-
-    this.setState({ emailError, hasEmptyPasswordError });
-
-    if (emailError && this.emailInputElement) {
+    const hasValidPhoneNumber = emailOrPhoneNumber
```

Comment:
There are much better ways to do these types of validations - ideally using `yup` or a similar tool to make our lives easier. I'm keeping it on the list of possible future improvements for when we get to the forms :)